jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/jdbc/JdbcAdapter.java

```diff
@@ -342,22 +347,21 @@ private static String getUserAndPasswordForImport(ExaMetadata exaMeta, SchemaMet
     }
 
     private static String createColumnDescription(final ExaMetadata exaMeta, final SchemaMetadataInfo meta,
-            final String pushdownQuery, final SqlDialect dialect) throws AdapterException {
-        PreparedStatement ps = null;
+            final String pushdownQuery, final SqlDialect dialect) {
         final ExaConnectionInformation connectionInformation = JdbcAdapterProperties
                 .getConnectionInformation(meta.getProperties(), exaMeta);
-
-        Connection connection = null;
         try {
-            connection = establishConnection(connectionInformation);
+            Connection connection = establishConnection(connectionInformation);
             logger.fine(() -> "createColumnDescription: " + pushdownQuery);
-            ps = connection.prepareStatement(pushdownQuery);
-            ResultSetMetaData metadata = ps.getMetaData();
-            if (metadata == null) {
-                ps.execute();
+            ResultSetMetaData metadata = null;
+            try (PreparedStatement ps = connection.prepareStatement(pushdownQuery)) {
                 metadata = ps.getMetaData();
                 if (metadata == null) {
-                    throw new SQLException("getMetaData() failed");
+                    ps.execute();
+                    metadata = ps.getMetaData();
+                    if (metadata == null) {
+                        throw new SQLException("getMetaData() failed");
```

Comment:
"Unable to read source metadata trying to create description for source columns."